### ISSUE TITLE
Fish on linux creates pipe ".notifier" in current directory when started without USER in env

### ISSUE THREAD

**Issue by chtison** [fish-shell/fish-shell#5180]
## Versions

x86_64 GNU/Linux
fish, version 2.7.1
$version = 2.7.1
$TERM = screen
$USER = root

## Behaviour

When fish is started without USER variable set in environment, it creates a pipe named ".notifier" in the working directory.

```sh
$ env --unset=USER fish
$ ls -l .notifier
prw------- 1 root root 0 Sep 11 11:25 .notifier|
```

The USER variable is correctly set but not soon enough.

```sh
$ env USER=rofl fish
$ ls -l /tmp/fish.rofl/
total 0
prw------- 1 root root 0 Sep 11 11:51 fishd.0242ac110004.notifier|
$ echo $USER
root
```

## Issue

https://github.com/fish-shell/fish-shell/blob/2.7.1/src/env_universal_common.cpp#L135-L137

The author of this code assumes USER is set but it is not at this time so function returns "" which is prepended to [".notifier"](https://github.com/fish-shell/fish-shell/blob/2.7.1/src/env_universal_common.cpp#L1399).

## Workaround

Start fish with USER variable correctly set in the env.

**Comment by zanchey** [comment]
That's definitely worth fixing, but what environment are you in that starts without USER set?

**Comment by chtison** [comment]
In a docker image where first process is tmux and fish is the login shell.
See last 2 lines of https://github.com/chtison/debian/blob/master/Dockerfile

**Comment by eblume** [comment]
I can see that this is resolved in 3.0 which I look forward to very much, but just to add to this as I was bit by it today:

I was investigating ways to have fish pick up configuration changes and tried `exec env -i fish` and `exec -c fish`. I quickly realized that this was not the way to go, but this pipe was created anyway. Others may find this report through google (as did I) for similar reasons, so to those future people: please upgrade to fish 3.0, which is not yet out (but should be out Nov 18 2018). Details here: https://github.com/fish-shell/fish-shell/milestone/18
